querydsl-codegen/src/main/java/com/querydsl/codegen/UncapitalizedCaseTransformer.java

```diff
@@ -0,0 +1,19 @@
+package com.querydsl.codegen;
+
+import com.mysema.codegen.StringUtils;
+
+/**
+ * Uncapitalizes the first letter of a name.
+ *
+ */
+public class UncapitalizedCaseTransformer implements CaseTransformer {
+
+    /**
+     * @see com.querydsl.codegen.CaseTransformer#transform(java.lang.String)
+     */
+    @Override
+    public String transform(String name) {
```

Comment:
Hi Ruben,

As per my last comment on the pull request, do you think the rest of the
code looks ok?
Should we inject the strategy class name in order to allow the user plug in
its own implementation?

Thanks,

Marcello

On 18 July 2015 at 15:35, Ruben Dijkstra <notifications@github.com> wrote:

> In
> querydsl-codegen/src/main/java/com/querydsl/codegen/UncapitalizedCaseTransformer.java
> <https://github.com/querydsl/querydsl/pull/1442#discussion_r34948186>:
>
> > @@ -0,0 +1,19 @@
> > +package com.querydsl.codegen;
> > +
> > +import com.mysema.codegen.StringUtils;
> > +
> > +/**
> > + * Uncapitalizes the first letter of a name.
> > + *
> > + */
> > +public class UncapitalizedCaseTransformer implements CaseTransformer {
> > +
> > +    /**
> > +     * @see com.querydsl.codegen.CaseTransformer#transform(java.lang.String)
> > +     */
> > +    @Override
> > +    public String transform(String name) {
>
> I think it's better to name this class 'DefaultVariableNameFunction' and
> make it a Guava Function.
> This makes the call from EntitySerializer a bit more natural and doesn't
> limit the Function to just String manipulation.
> What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/querydsl/querydsl/pull/1442/files#r34948186>.
>